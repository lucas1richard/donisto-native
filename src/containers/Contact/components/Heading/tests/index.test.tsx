import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ContactHeading, { IContactHeadingProps } from '..';

describe('<ContactHeading />', () => {
  const renderComponent = (props: IContactHeadingProps) => toJson(shallow(<ContactHeading {...props} />));
  it('renders with just text', () => {
    expect(renderComponent({
      contact: {
        city: 'mockCity',
        state: 'mockState',
        first_name: 'mockFirstName',
        last_name: 'mockLastName',
      }
    })).toMatchSnapshot();
  });
});