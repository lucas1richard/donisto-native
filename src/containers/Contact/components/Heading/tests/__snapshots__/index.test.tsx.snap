// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactHeading /> renders with just text 1`] = `
<Component
  style={
    Object {
      "height": 200,
      "marginBottom": 120,
      "position": "relative",
    }
  }
>
  <ImageBackground
    imageStyle={
      Object {
        "alignSelf": "flex-start",
        "flex": 1,
        "height": 200,
        "overflow": "hidden",
      }
    }
    source={1}
    style={
      Object {
        "backgroundColor": "#000",
        "flexDirection": "row",
        "height": 200,
        "justifyContent": "flex-end",
        "paddingHorizontal": 12,
        "paddingVertical": 12,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Component>
        <Icon
          allowFontScaling={false}
          color="#fff"
          name="dots-three-horizontal"
          size={40}
        />
      </Component>
    </TouchableOpacity>
  </ImageBackground>
  <Component
    style={
      Object {
        "backgroundColor": "#dedede",
        "borderColor": "#fff",
        "borderRadius": 20,
        "borderWidth": 6.666666666666667,
        "bottom": -120,
        "height": 200,
        "left": 12,
        "position": "absolute",
        "width": 133.33333333333331,
        "zIndex": 50,
      }
    }
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Component
      style={
        Object {
          "bottom": 12,
          "maxWidth": 300,
          "minWidth": 100,
          "position": "absolute",
          "right": 12,
          "zIndex": 50,
        }
      }
    >
      <Txt
        color="light"
      >
        mockFirstName
         
        mockLastName
      </Txt>
      <Txt
        color="light"
      >
        mockCity, mockState
      </Txt>
    </Component>
  </TouchableOpacity>
</Component>
`;
